client/app/cloud/project/compute/infrastructure/virtualMachine/add/cloud-project-compute-infrastructure-virtualMachine-add.service.js

```diff
@@ -0,0 +1,181 @@
+class CloudProjectVirtualMachineAddService {
+    constructor ($q, $translate, CloudFlavorService, CloudImageService, ControllerModalHelper, OvhApiCloudProject, OvhApiCloudProjectInstance, OvhApiCloudProjectNetworkPrivateSubnet) {
+        this.$q = $q;
+        this.$translate = $translate;
+        this.CloudFlavorService = CloudFlavorService;
+        this.CloudImageService = CloudImageService;
+        this.ControllerModalHelper = ControllerModalHelper;
+        this.OvhApiCloudProject = OvhApiCloudProject;
+        this.OvhApiCloudProjectInstance = OvhApiCloudProjectInstance;
+        this.OvhApiCloudProjectNetworkPrivateSubnet = OvhApiCloudProjectNetworkPrivateSubnet;
+    }
+
+    getAugmentedImages (images) {
+        return _.map(_.uniq(images, "id"), this.CloudImageService.augmentImage);
+    }
+
+    getAugmentedFlavorsFilteredByType (flavors, type) {
+        return _.filter(_.map(_.filter(flavors, {
+            available: true,
+            osType: type
+        }), flavor => this.CloudFlavorService.augmentFlavor(flavor)), {
+            diskType: "ssd",
+            flex: false
+        });
+    }
+
+    getFilteredFlavorsByRegion (flavors, regionCode) {
+        const filteredFlavors = _.uniq(_.remove(flavors, { region: regionCode }), "name");
+        const usedFlavorNames = _.uniq(_.map(filteredFlavors, flavor => flavor.name));
+        const notAvailableFlavors = _.filter(flavors, flavor => !_.include(usedFlavorNames, flavor.name));
+        const outOfRegionFlavors = _.map(_.uniq(notAvailableFlavors, "name"), flavor => {
+            flavor.regions = _.map(_.filter(notAvailableFlavors, f => f.name === flavor.name), "region");
+            flavor.disabled = "NOT_AVAILABLE";
+            delete flavor.region;
+            delete flavor.price;
+            return flavor;
+        });
+
+        return filteredFlavors.concat(outOfRegionFlavors);
+    }
+
+    getFilteredPrivateNetworksByRegion (privateNetworks, regionCode, subNets = []) {
+        return _.chain(privateNetworks)
+            .filter(network => {
+                if (!_.has(subNets, network.id)) {
+                    return false;
+                }
+                return _.some(network.regions, "region", regionCode);
+            })
+            .sortBy("vlanId")
+            .map(network => {
+                const pad = Array(5).join("0");
+                return _.assign(network, {
+                    vlanId: pad.substring(0, pad.length - network.vlanId.toString().length) + network.vlanId
```

Comment:
_.padLeft _.padRight
or native padStart / padEnd (might need a polyfill for IE tho)